Dockerfile

```diff
@@ -1,32 +1,38 @@
-FROM jupyter/base-notebook:python-3.7.6
-
+FROM jupyter/base-notebook
 
 USER root
 
-RUN apt-get -y update \
- && apt-get install -y dbus-x11 \
-   firefox \
-   xfce4 \
-   xfce4-panel \
-   xfce4-session \
-   xfce4-settings \
-   xorg \
-   xubuntu-icon-theme
+RUN apt-get -y update > /dev/null \
+ && apt-get -y upgrade > /dev/null \
+ && apt-get -y install \
+        dbus-x11 \
+        firefox \
+        xfce4 \
+        xfce4-panel \
+        xfce4-session \
+        xfce4-settings \
+        xorg \
+        xubuntu-icon-theme \
+    > /dev/null \
+    # chown $HOME to workaround that the xorg installation creates a
+    # /home/jovyan/.cache directory owned by root
+ && chown -R $NB_UID:$NB_GID $HOME \
+ && rm -rf /var/lib/apt/lists/*
 
-# Remove light-locker to prevent screen lock
+# Install TurboVNC (https://github.com/TurboVNC/turbovnc)
 ARG TURBOVNC_VERSION=2.2.6
-RUN wget -q "https://sourceforge.net/projects/turbovnc/files/${TURBOVNC_VERSION}/turbovnc_${TURBOVNC_VERSION}_amd64.deb/download" -O turbovnc_${TURBOVNC_VERSION}_amd64.deb && \
-   apt-get install -y -q ./turbovnc_${TURBOVNC_VERSION}_amd64.deb && \
-   apt-get remove -y -q light-locker && \
-   rm ./turbovnc_${TURBOVNC_VERSION}_amd64.deb && \
-   ln -s /opt/TurboVNC/bin/* /usr/local/bin/
-
-# apt-get may result in root-owned directories/files under $HOME
-RUN chown -R $NB_UID:$NB_GID $HOME
+RUN wget -q "https://sourceforge.net/projects/turbovnc/files/${TURBOVNC_VERSION}/turbovnc_${TURBOVNC_VERSION}_amd64.deb/download" -O turbovnc.deb \
+ && apt-get install -y ./turbovnc.deb > /dev/null \
+    # remove light-locker to prevent screen lock
+ && apt-get remove -y light-locker > /dev/null \
```

Comment:
```suggestion
 && apt-get install -y -q ./turbovnc.deb \
    # remove light-locker to prevent screen lock
 && apt-get remove -y -q light-locker \
```